src/test/java/com/epam/tc/hw7/JdiTest.java

```diff
@@ -0,0 +1,102 @@
+package com.epam.tc.hw7;
+
+import static com.epam.jdi.light.elements.init.PageFactory.initElements;
+import static com.epam.tc.hw7.pages.HomePage.loginButton;
+import static com.epam.tc.hw7.pages.HomePage.password;
+import static com.epam.tc.hw7.pages.HomePage.user;
+import static com.epam.tc.hw7.pages.HomePage.userIcon;
+import static com.epam.tc.hw7.pages.HomePage.userName;
+import static com.epam.tc.hw7.pages.JdiSite.homePage;
+import static com.epam.tc.hw7.pages.MetalsPage.checkboxesLabels;
+import static com.epam.tc.hw7.pages.MetalsPage.colorButton;
+import static com.epam.tc.hw7.pages.MetalsPage.colorResult;
+import static com.epam.tc.hw7.pages.MetalsPage.colorsDrop;
+import static com.epam.tc.hw7.pages.MetalsPage.elementsResult;
+import static com.epam.tc.hw7.pages.MetalsPage.metalButton;
+import static com.epam.tc.hw7.pages.MetalsPage.metalsDrop;
+import static com.epam.tc.hw7.pages.MetalsPage.metalsResult;
+import static com.epam.tc.hw7.pages.MetalsPage.radioButton;
+import static com.epam.tc.hw7.pages.MetalsPage.submitButton;
+import static com.epam.tc.hw7.pages.MetalsPage.summaryResult;
+import static com.epam.tc.hw7.pages.MetalsPage.vegetablesButton;
+import static com.epam.tc.hw7.pages.MetalsPage.vegetablesDrop;
+import static com.epam.tc.hw7.pages.MetalsPage.vegetablesResult;
+
+import com.epam.jdi.light.driver.WebDriverUtils;
+import com.epam.tc.hw7.pages.JdiSite;
+import com.epam.tc.hw7.pages.MetalsPage;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class JdiTest {
+
+    @BeforeSuite(alwaysRun = true)
+    public void beforeSuite() {
+        initElements(JdiSite.class);
+        homePage.open();
+    }
+
+    @AfterSuite(alwaysRun = true)
+    public void afterSuite() {
+        WebDriverUtils.killAllSeleniumDrivers();
+    }
+
+    @Test(dataProvider = "jsonData")
+    public void loginTest(Map<String, MetalsPageTestData> data) {
+        userIcon.click();
+        user.sendKeys("Roman");
+        password.sendKeys("Jdi1234");
+        loginButton.click();
+        userName.assertThat().displayed();
+        JdiSite.metalsPage.open();
```

Comment:
It's better than it was before. But still too many actions in the test itself. Hide form filling and login in separate methods, which you will use here. Create separate classes for forms, inside which these methods will be.

Such test is difficult to read and maintain. It will be much easier if there are steps like 1. login, 2. fill form, 3. check something